zeth_tool/split_keypair_cmd.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2015-2021 Clearmatics Technologies Ltd
+//
+// SPDX-License-Identifier: LGPL-3.0+
+
+#include "zeth_tool/split_keypair_cmd.hpp"
+
+#include "libtool/tool_util.hpp"
+
+namespace zethtool
+{
+
+namespace commands
+{
+
+class split_keypair_cmd : public generic_subcommand<split_keypair_cmd>
+{
+public:
+    using base_class = generic_subcommand<split_keypair_cmd>;
+
+    split_keypair_cmd(
+        const std::string &subcommand_name, const std::string &description)
+        : base_class(subcommand_name, description)
+    {
+    }
+
+    template<typename ppT, typename snarkT>
+    int execute_generic(const global_options &)
+    {
+        ppT::init_public_params();
+        libff::inhibit_profiling_info = true;
+        libff::inhibit_profiling_counters = true;
+
+        typename snarkT::keypair keypair;
+        {
+            std::ifstream in_s = libtool::open_binary_input_file(keypair_file);
+            snarkT::keypair_read_bytes(keypair, in_s);
+        }
+
+        if (!vk_file.empty()) {
+            std::ofstream out_s = libtool::open_binary_output_file(vk_file);
+            snarkT::verification_key_write_bytes(keypair.vk, out_s);
+        }
+
+        if (!pk_file.empty()) {
+            std::ofstream out_s = libtool::open_binary_output_file(pk_file);
+            snarkT::proving_key_write_bytes(keypair.pk, out_s);
+        }
+
+        return 0;
+    }
+
+protected:
+    void initialize_suboptions(
+        boost::program_options::options_description &options,
+        boost::program_options::options_description &all_options,
+        boost::program_options::positional_options_description &pos) override
+    {
+        base_class::initialize_suboptions(options, all_options, pos);
+
+        options.add_options()(
+            "vk-file,v",
+            po::value<std::string>(),
+            "Verification key file (optional)")(
+            "pk-file,p",
+            po::value<std::string>(),
+            "Proving key file (optional)");
+
+        all_options.add(options).add_options()(
+            "keypair_file,k", po::value<std::string>(), "Keypair file");
```

Comment:
Use of snake_case in `keypair_file` is inconsistent with the use of `-` above (i.e. `pk-file` and `vk-file`), so let's switch to `keypair-file`

Reply:
renamed the argument key